### PR TITLE
Lecture 23

### PR DESCRIPTION
**Pull request by snasiriany**

### CODE REVIEW

File: notes/lecture23.tex [hunk 1]
```diff
@@ -0,0 +1,134 @@
+\section{Lecture 23: Newton's Method}
+Up until now, we have only considered first order  methods to optimize functions. Now, we will utilize second order information to achieve a faster rate of convergence.
+\\
+\\
```
Comment (mrtzh) [inline notes/lecture23.tex]: Can you please remove all such manual line breaks.

File: notes/lecture23.tex [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+To illustrate the point, consider a single variable function $\varphi:\R \rightarrow \R$. Our goal is to solve the non-linear equation $\varphi(x) = 0$. From Taylor's theorem, we can express the first-order form of $\varphi(x)$ as
+$$\varphi(x) = \varphi(x_0) + \varphi'(x) \cdot (x-x_0) + o(\abs{x-x_0})$$
+given $\delta = x - x_0$ we equivalently have that
+$$\varphi(x_0 + \delta) = \varphi(x_0) + \varphi'(x) \cdot \delta + o(\abs{\delta})$$
+Disregarding the $o(\abs{\delta})$ term, we solve (over $\delta$) the following objective: $$\varphi(x_0) + \varphi'(x_0) \delta = 0$$
+Then, $\delta = -\frac{\varphi(x_0)}{\varphi'(x_0)}$, leading to the iteration 
+$x_{t+1} = x_t - \frac{\varphi(x_t)}{\varphi'(x_t)}$.
+\\
+\\
+We cam similarly make an argument for a multi variable function $F: \R^d \rightarrow \R$. Our goal is to solve $F(x) = 0$. Again, from Taylor's theorem we have that
```
Comment (mrtzh) [inline notes/lecture23.tex]: can

File: notes/lecture23.tex [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+   \end{cases}
+\]
+Note the similarity between underdamped, critically damped, and overdamped dynamical systems. This example shows that even for strongly convex functions with Lipschitz gradients that Newton's method is only guaranteed to converge locally. To avoid divergence, a popular technique is to use a \textit{damped} step--size:
+\begin{align*}
+    x_{t+1} = x_t - \eta_t \nabla^2 f(x_t)^{-1} \nabla f(x_t)
+\end{align*}
+$\eta_t$ can be chosed by backtracking line search. Usually though $\eta = 1$ is a good first choice since, if you are in a region of convergence, you are guaranteed quadratic convergence.
+
+\subsection{Quasi--Newton Methods}
+Comparing Gradient Descent and Newton's method side by side:
```
Comment (mrtzh) [inline notes/lecture23.tex]: Please lower case gradient descent.

File: notes/lecture23.tex [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    \item Gradient Descent
+    \begin{align*}
+        x_{t+1} = x_t - \eta_t \nabla f(x_t)
+    \end{align*}
+    \item Newton's Method
+    \begin{align*}
+        x_{t+1} = x_t - \nabla^2 f(x_t)^{-1} \nabla f(x_t)
+    \end{align*}
+\end{enumerate}
+we see that gradient descent approximates $\nabla^2 f(x_t)^{_1}$ as a scaled version of the identity. That is, gradient descent is equivalent to Newton's method when $\nabla^2 f(x_t)^{-1} = \eta_t I$ where $I$ is the identity matrix. Quasi--newton methods take the analogy a step further by approximating the Hessian by some other matrix. The idea in doing so is to avoid an expensive matrix inversion at each step. What we want is an approximation
```
Comment (mrtzh) [inline notes/lecture23.tex]: We

File: notes/lecture23.tex [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      |x_t| > 1 & \text{Algorithm diverges} 
+   \end{cases}
+\]
+Note the similarity between underdamped, critically damped, and overdamped dynamical systems. This example shows that even for strongly convex functions with Lipschitz gradients that Newton's method is only guaranteed to converge locally. To avoid divergence, a popular technique is to use a \textit{damped} step--size:
+\begin{align*}
+    x_{t+1} = x_t - \eta_t \nabla^2 f(x_t)^{-1} \nabla f(x_t)
+\end{align*}
+$\eta_t$ can be chosed by backtracking line search. Usually though $\eta = 1$ is a good first choice since, if you are in a region of convergence, you are guaranteed quadratic convergence.
+
+\subsection{Quasi--Newton Methods}
```
Comment (mrtzh) [inline notes/lecture23.tex]: use single dash for hyphen

File: notes/lecture23.tex (hunk 4 above)
Comment (mrtzh) [reply notes/lecture23.tex]: ^{_1} should be ^{-1}

File: notes/lecture23.tex [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    \item Newton's Method
+    \begin{align*}
+        x_{t+1} = x_t - \nabla^2 f(x_t)^{-1} \nabla f(x_t)
+    \end{align*}
+\end{enumerate}
+we see that gradient descent approximates $\nabla^2 f(x_t)^{_1}$ as a scaled version of the identity. That is, gradient descent is equivalent to Newton's method when $\nabla^2 f(x_t)^{-1} = \eta_t I$ where $I$ is the identity matrix. Quasi--newton methods take the analogy a step further by approximating the Hessian by some other matrix. The idea in doing so is to avoid an expensive matrix inversion at each step. What we want is an approximation
+\begin{align*}
+    \hat{f}_{B_t}(x) \approx f(x_t) + \nabla f(x_t)^\trans (x - x_t) + \dfrac{1}{2} (x-x_t)B_t^{-1} (x-x_t)
+\end{align*}
+such that
```
Comment (mrtzh) [inline notes/lecture23.tex]: such that:

File: notes/lecture23.tex [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+\begin{enumerate}
+    \item $\nabla \hat{f}_{B_t} (x_t) = \nabla f(x_t)$.\\
+    It seems reasonable that our approximation should be the same up to first order. 
+    \item $\nabla \hat{f}_{B_t} (x_{t-1}) = \nabla f(x_{t-1})$\\
+    This condition states that the gradient should still be correct at the previous iterate.
+\end{enumerate}
+ If the two last gradients are correct, we can expect our Hessian approximation to be reasonable along the direction $x_t - x_{t-1}$. This is called a \emph{Secant Approximation} which can be written as
```
Comment (mrtzh) [inline notes/lecture23.tex]: lower case secant approximation

**Comment by snasiriany** [comment]
Thanks for the feedback! Changes have been implemented.

**Pull request merged**